packages/query-core/src/queryObserver.ts

```diff
@@ -349,7 +348,7 @@ export class QueryObserver<
     )
 
     if (!fetchOptions?.throwOnError) {
-      promise = promise.catch(noop)
+      promise = promise.catch(() => undefined)
```

Comment:
why can’t we continue to use `noop` here ?

Reply:
If we use noop(function noop(): void {}, not function noop(): undefined {}), TypeScript error will be occured

Reply:
at runtime, there’s no difference between a function that returns `undefined` and a function that returns `void`. would be good if we could keep `noop` here

Reply:
I agree, I reflect your review now in [92b4171](https://github.com/TanStack/query/pull/8294/commits/92b4171b1a7fdf5f4c179798201bb6b608bbd572)

Reply:
nice 👍

Reply:
hmm TypeScript doesn’t like that :/